packages/blocks/src/submit-button/index.js

```diff
@@ -1,13 +1,22 @@
 /**
  * Internal dependencies
  */
-import { Button } from '../components';
+import { Button, ButtonSpinner } from '../components';
+import { useFormState } from '@crowdsignal/form';
 
 const SubmitButton = ( { attributes, className } ) => {
+	const { isSubmitting } = useFormState();
+
+	const RenderedButton = isSubmitting ? ButtonSpinner : Button;
+
 	return (
-		<Button attributes={ attributes } className={ className } type="submit">
+		<RenderedButton
+			attributes={ attributes }
+			className={ className }
+			type="submit"
```

Comment:
We should set `disabled={ isSubmitting }` as otherwise it's still possible to re-submit a form that's already being submitted which may or may not cause any weird side effects.

The reason we use `pointer-events: none;` in the editor over `disabled` is it's important to have a good idea what the input will look like when you publish the project - which the `disabled` state would affect.  
But in this case, on the public end, I'd say this no longer applies and if things need be disabled, they should look like they're disabled.

Reply:
Got it! Thanks for explaining it.